src/lib/optimizer/chunk_statistics.hpp

```diff
@@ -0,0 +1,218 @@
+#pragma once
+
+#include <exception>
+#include <memory>
+#include <type_traits>
+#include <vector>
+
+#include "all_type_variant.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+#include "storage/base_encoded_column.hpp"
+#include "storage/deprecated_dictionary_column.hpp"
+#include "storage/dictionary_column.hpp"
+#include "storage/reference_column.hpp"
+#include "storage/run_length_column.hpp"
+#include "storage/value_column.hpp"
+
+namespace opossum {
+
+class BaseFilter : public std::enable_shared_from_this<BaseFilter> {
+ public:
+  virtual ~BaseFilter() = default;
+
+  virtual bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const = 0;
+};
+
+class ChunkColumnStatistics {
+ public:
+  static std::shared_ptr<ChunkColumnStatistics> build_statistics(DataType data_type, std::shared_ptr<BaseColumn> column);
+
+  void add_filter(std::shared_ptr<BaseFilter> filter) {
+    _filters.emplace_back(filter);
+  }
+
+  bool can_prune(const AllTypeVariant& value, const PredicateCondition predicate_type) const {
```

Comment:
Not happy with the "can_prune" name - it's not bad but I'd prefer something that says "can predicate match any tuples in this Chunk?"

Reply:
Or `do_tuples_qualify_predicate`, `do_tuples_qualify`?

Reply:
the logic is reversed, if can_prune returns `true` the chunk can be pruned. maybe something like "predicate does not yield tuples"?